### PR TITLE
Changes on webcam skip preview behavior

### LINKED ISSUE #10454: 'userdata-bbb_skip_video_preview=true' isn't affecting the meeting
tested in **2.3 servers**

**to reproduce**, start a meeting with the following custom parameter: **`userdata-bbb_skip_video_preview=true`**
and start sharing webcam:

**expectation**: 
webcam preview modal should not appear

**test result**:
webcam preview modal is appearing

### PR DESCRIPTION
**Pull request by ramonlsouza**
### What does this PR do?

Reverts the changes in PR #8353, so if the `userdata-bbb_skip_video_preview` parameter is set to `true` the video preview always skips
Introduces `userdata-bbb_skip_video_preview_on_first_join` parameter to skip only on first joining of webcam

### Closes Issue(s)

closes #10454

### CODE REVIEW

**Comment by mabras** [comment]
I have the issue https://github.com/bigbluebutton/bigbluebutton/issues/10454 on 2.2.31.
Your PR is targeting 2.3 only. Is there a change you can fix it for 2.2-realse?

Thanks anyway,

File: bigbluebutton-html5/imports/ui/components/video-provider/service.js [hunk 1]
```diff
@@ -577,12 +581,19 @@ class VideoService {
...
+    if (this.skipVideoPreviewOnFirstJoin === null) {
+      this.skipVideoPreviewOnFirstJoin = getFromUserSettings('bbb_skip_video_preview_on_first_join', false) || SKIP_VIDEO_PREVIEW_ON_FIRST_JOIN;
+    }
     if (this.skipVideoPreview === null) {
       this.skipVideoPreview = getFromUserSettings('bbb_skip_video_preview', false) || SKIP_VIDEO_PREVIEW;
     }
 
-    return this.skipVideoPreview && !fromInterface;
+    if(this.isFirstJoin){
+      return this.skipVideoPreviewOnFirstJoin || this.skipVideoPreview;
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/video-provider/service.js:593]: You could simplify this whole block to
`return (this.isFirstJoin && this.skipVideoPreviewOnFirstJoin) || this.skipVideoPreview;

Comment (antobinary) [reply bigbluebutton-html5/imports/ui/components/video-provider/service.js:593]: Also please lint your new code (only)

**Review by antobinary — CHANGES_REQUESTED** [review]
I tested locally and it worked well.

I see that  you keep the state for `isFirstJoin` in the video provider, which causes slight inconsistency with what we do for audio in #11284 specifically after a refresh. The session variable there persists while the state in video-provider is wiped.

Would you be alright switching to Session.set to match the behaviour?
If you have strong opinion on having audio check adjusted to match your implementation, now is the time to voice out your opinion ;)

**Comment by prlanzarin** [comment]
> I tested locally and it worked well.
> 
> I see that you keep the state for `isFirstJoin` in the video provider, which causes slight inconsistency with what we do for audio in #11284 specifically after a refresh. The session variable there persists while the state in video-provider is wiped.
> 
> Would you be alright switching to Session.set to match the behaviour?
> If you have strong opinion on having audio check adjusted to match your implementation, now is the time to voice out your opinion ;)

I'd prefer if it was not included in video-provider service's trackable props; instead use the usual pattern (`Session`) OR make it a regular class attribute instead of including it in the defineProperties proxy (if you do not wish to persist the behaviour across refreshes).

That collection of properties is conventionally reserved only for things that need to trigger modifications to the stream's list processing, which is not the case of the preview skip.

I also suggest, while you're at it, a few more things:
  - Move that whole skip shenanigan to video-preview's service. That code being in video-provider is a conflict of roles.
  - There's no need for the skipEtc to be a trackable props in video-preview's container. AFAIK those are not _volatile_ props per se: once you open the video-preview modal, you probe for it's return value. The return value of the skip checks only changes after you share a camera, which implies an unmount of the video-preview component. That implies that the skip checks will not change while the modal is open AFAIK.

**Comment by ramonlsouza** [comment]
Thanks for the review @antobinary and @prlanzarin! Implemented both suggestions.

**Review by antobinary — APPROVED** [review]
Looks good to me!
@prlanzarin any additional comments or we merge?

**Comment by prlanzarin** [comment]
I think skipVideoPreview still doesn't need to be a prop bolted in video-preview's' container (which uses a tracker). If possible, we'd better just call it directly (VideoPreviewService.getEtc) from inside the component if that works. Not blocking, though.

**Comment by antobinary** [comment]
Documented via https://github.com/bigbluebutton/bigbluebutton.github.io/pull/231

**Pull request merged**